### DISCUSSION TITLE
Reformat MAC address for custom link

### DISCUSSION THREAD

**Discussion by Azmodeszer** [netbox-community/netbox#15128] [Q&A] [answered]
I'm looking for a way to preprocess data for a custom link, specifically the MAC address.

The plan is to check if an object has a primary IPv4 set and then check if the corresponding interface has a MAC address. If it does, take that MAC address, let's say AA:BB:CC:DD:EE:FF and turn it into

`%22AA%3ABB%3ACC%3ADD%3AEE%3AFF%22`

which I can then insert into the custom link to an external FDB table

`https://server.local/search/address={{formatted_mac}}/deleted=0/search=fdb/`

The logic and syntax for selecting the MAC in the first place I can probably figure out on my own, but transforming it so that it results in a working link isn't immediately obvious. Doesn't seem like it's possible directly in the Jinja2 template field?

Thanks.

**Comment by candlerb** [comment, marked as answer]
You want the [jinja2 template designer documentation](https://jinja.palletsprojects.com/en/3.1.x/templates/), especially [built-in filters](https://jinja.palletsprojects.com/en/3.1.x/templates/#list-of-builtin-filters).

Your example looks a bit weird, but I think you could do

```
https://server.local/search/address=%2A{{ interface.mac_address | replace(":","%3A") }}%2A/deleted=0/search=fdb/
```

**Reply by Azmodeszer** [reply]
Well damn, much simpler than anticipated. Thanks a bunch.
